metadetect/mbobs_list_fitting.py

```diff
@@ -0,0 +1,369 @@
+        else 0
+    )
+
+    # now we combine via inverse variance weighting
+    tot_res = np.zeros(
+        len(mbobs_list),
+        dtype=_make_fitter_mbobs_sep_dtype(nband, nonshear_nband, fitter.model),
+    )
+    for ind in range(len(mbobs_list)):
+        wgts, all_bres = _extract_band_ind_res_fitter_mbobs_sep(
+            ind, mbobs_list, nonshear_mbobs_list, band_res, nband, nonshear_nband
+        )
+        shear_mbobs = mbobs_list[ind]
+        nonshear_mbobs = (
+            nonshear_mbobs_list[ind]
+            if nonshear_mbobs_list is not None
+            else None
+        )
+        tot_res[ind] = _compute_wavg_fitter_mbobs_sep(
+            wgts, all_bres, all_is_shear_band,
+            shear_mbobs, nband, nonshear_nband,
+            fitter.model,
+            nonshear_mbobs=nonshear_mbobs,
+        )
+
+    if len(tot_res) == 0:
+        return None
+    else:
+        return tot_res
+
+
+def _run_fitter_mbobs_sep_go(
+    *,
+    mbobs_list,
+    fitter,
+    nonshear_mbobs_list,
+):
+    nband = len(mbobs_list[0])
+    band_res = []
+    all_is_shear_band = []
+
+    for band in range(nband):
+        all_is_shear_band.append(True)
+        band_mbobs_list = []
+        for mbobs in mbobs_list:
+            _mbobs = ngmix.MultiBandObsList()
+            _mbobs.meta = copy.deepcopy(mbobs.meta)
+            _mbobs.append(mbobs[band])
+            band_mbobs_list.append(_mbobs)
+        band_res.append(fitter.go(band_mbobs_list))
+
+    if nonshear_mbobs_list is not None:
+        nonshear_nband = len(nonshear_mbobs_list[0])
+        for band in range(nonshear_nband):
+            all_is_shear_band.append(False)
+            band_mbobs_list = []
+            for mbobs in nonshear_mbobs_list:
+                _mbobs = ngmix.MultiBandObsList()
+                _mbobs.meta = copy.deepcopy(mbobs.meta)
+                _mbobs.append(mbobs[band])
+                band_mbobs_list.append(_mbobs)
+            band_res.append(fitter.go(band_mbobs_list))
+
+    return band_res, all_is_shear_band
+
+
+def _make_fitter_mbobs_sep_dtype(nband, nonshear_nband, model):
+    # combine the data by inverse variance weighted averages
+    tot_nband = nband + nonshear_nband
+    n = Namer(front=model)
+    dt = [
+        ("flags", 'i4'),
+        (n("flags"), 'i4'),
+        (n("s2n"), "f8"),
+        (n("T"), "f8"),
+        (n("T_err"), "f8"),
+        (n("g"), "f8", 2),
+        (n("g_cov"), "f8", (2, 2)),
+        ('psf_g', 'f8', 2),
+        ('psf_T', 'f8'),
+        (n("T_ratio"), "f8"),
+    ]
+    if tot_nband > 1:
+        dt += [
+            (n("band_flux_flags"), 'i4'),
+            (n("band_flux"), "f8", tot_nband),
+            (n("band_flux_err"), "f8", tot_nband),
+        ]
+    else:
+        dt += [
+            (n("band_flux_flags"), 'i4'),
+            (n("band_flux"), "f8"),
+            (n("band_flux_err"), "f8"),
+        ]
+    return dt
+
+
+def _extract_band_ind_res_fitter_mbobs_sep(
```

Comment:
what does the `_ind` mean in this function name?  Why is `_fitter` needed in the name?

Could it more clearly by called  `_extract_band_results_sep` or something like that?

Reply:
If the worry is that the name is too generic, I think it can be argued that a better way to namespace is through the module, e.g.

`module.function` rather than `specific_prefix_function`

in other words, import the module rather than import the function from the module.

If this is too much bikeshedding then feel free to ignore